### ISSUE TITLE
Very long activation time

### ISSUE THREAD

**Issue by ziki77** [Serhioromano/vscode-gitflow#23]
Hi,
the extension is good, but it takes a lot of time to activate:

![image](https://user-images.githubusercontent.com/10797548/161217935-799f4e8a-59ed-462c-a74b-2eddcca960d3.png)

Is this normal? Does it depend by the repo complexity?

Thanks

**Comment by Serhioromano** [comment]
This extension requires Git and Git-base core extensions you can see it in extension details page. It means that extension will not start activation until SCM is loaded. So, first after you launch VS Code it waits until SCM section loads and then it loads itself.

After CSM is loaded it might take a second, but after VS Code launch it will be definitely longer.

Anyway, thank you for nothing. If you like this extension, I'll appreciate if you rate it here
https://marketplace.visualstudio.com/items?itemName=Serhioromano.vscode-gitflow&ssr=false#review-details

**Comment by ziki77** [comment]
Sorry, but with Git Flow extension disabled the SCM section loads in half the time (7sec vs 14sec).

**Comment by Serhioromano** [comment]
What 14s?

**Comment by ziki77** [comment]
Yes but it is not the absolute value the problem, it is the difference. If it should take 1 sec more after the SCM is loaded it should be 8sec.
For SCM loaded I mean from when I launch VSC to when I can see the "Changes" in SCM.

**Comment by Serhioromano** [comment]
I'll try to check a performance.

**Comment by Serhioromano** [comment]
I've created a small test with time measurements. It turns out that delay is command `git remote` to get list of all remotes. I think this is the problem. But it is essential for Git Flow to know if ORIGIN exists. It is better to have it once on load than every time before command.

I'll think how I can optimize this.

**Comment by Serhioromano** [comment]
Also if ORIGIN exists it pulls all tags from it

    "git" ls-remote --tags origin

Also all branches

    "git" branch -r

**Comment by ziki77** [comment]
And it depends on how many branches, tags, ecc the repo has, I think. I can try with a very simple repo.

**Comment by Serhioromano** [comment]
It might also depend on your connection speed. Give me 3 minutes I'll release new version where in the log you can see time what took most of the time.

**Comment by Serhioromano** [comment]
https://github.com/Serhioromano/vscode-gitflow/actions
Building right now.

**Comment by ziki77** [comment]
> It might also depend on your connection speed. Give me 3 minutes I'll release new version where in the log you can see time what took most of the time.

Mmmm yes today I work from home through VPN, I will try next time in the office.

**Comment by Serhioromano** [comment]
Published. Give marketplace 2-3 minutes to approve. Version 1.3.3. Open OUTPUT and Git Flow stream, there should a time for every git command. Tell me what takes time.

**Comment by ziki77** [comment]
OK, thanks I will try and give you a feedback here.

**Comment by Serhioromano** [comment]
Any result?

**Comment by Serhioromano** [comment]
Do not forget to rate. I need it badly. Very hard to get ranged in the beginning.

**Comment by ziki77** [comment]
Already done ;)

**Comment by ziki77** [comment]
Hi,
I tried in the office so in the same LAN of the git server. Without the extension it take 3sec to load SCM, with the extension enabled like from home after the latest fix (8-9sec).

**Comment by Serhioromano** [comment]
Hmmmm, I am going to look into it when I come back home from trip.

**Comment by Serhioromano** [comment]
> Due to gitflow extension my git repo detection under git tab takes very log time ~10s

I think it is because Git Flow also fetch remote branches if your repo have long history or many files it might take few seconds.

**Comment by osvaldont** [comment]
My gitflow takes a long time, in operations like creating start feature, start hotfix, I'm windows. Any suggestions on how to improve performance.

**Comment by Serhioromano** [comment]
> My gitflow takes a long time, in operations like creating start feature, start hotfix, I'm windows. Any suggestions on how to improve performance.

Do you work in WSL?

**Comment by osvaldont** [comment]
No. I'm on windows.

**Comment by Serhioromano** [comment]
> No. I'm on windows.

What is your repository? Is the size is big? You have HDD or SSSD?

**Comment by osvaldont** [comment]
My repository is small, it has only one class. My PC is SSD. To make a git flow init, it takes 6 minutes

**Comment by osvaldont** [comment]
Ha a delay too when you switch branches

**Comment by Serhioromano** [comment]
You mean delay when you run `git flow init` in a terminal? Then it is not related to my extension. It is a delay by GitFLow AVH

**Comment by osvaldont** [comment]
You're right. When I run git commands it's super fast, but when I run commands from the git flow extension, it's very slow. Any tips to make it faster?

**Comment by Serhioromano** [comment]
> You're right. When I run git commands it's super fast, but when I run commands from the git flow extension, it's very slow. Any tips to make it faster?

Git commands or gitflow commands?

**Comment by diverger** [comment]
I find this problem too. In Startup performance of VSCode, the 'active call' show 1804, that means it take 18s to startup?!!

**Comment by Serhioromano** [comment]
Time passed and now we have tools we did not have before. I made lot of improvements to this issue. Please anyone test 1.5.0 and say is it faster now?
